### ISSUE TITLE
Rebuilding 2.0 for JDK6?

### ISSUE THREAD

**Issue by saper** [msteiger/jxmapviewer2#36]
Hi, I just noticed that the distributed [jxmapviewer2-2.0.jar] (http://search.maven.org/remotecontent?filepath=org/jxmapviewer/jxmapviewer2/2.0/jxmapviewer2-2.0.jar) on the Maven Central corresponds to 159950c94b7d27b42d1aad867404e47f0bf6707a, and to to tag `2.0` (which is 503b5e46958ce45251b6163cd13dadbc26cdefb5). Most importantly, the images have been moved.

I managed to compile 503b5e46958ce45251b6163cd13dadbc26cdefb5 using maven and got the jar (unfortunately with strange paths like `main/...` that made it into the release.

However I am stuck at the newer `gradle` process:

Using `gradle`, I get:

````
* Where:
Build file '/home/saper/sw/jxmapviewer2/jxmapviewer2/build.gradle' line: 27

* What went wrong:
A problem occurred evaluating root project 'jxmapviewer2'.
> No such property: sonatypeRepoURI for class: org.gradle.api.publication.maven.internal.ant.DefaultGroovyMavenDeployer

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 13.241 secs
````

When I used Maven on 503b5e46958ce45251b6163cd13dadbc26cdefb5 I managed to compile the code for a JDK6 successfully.

Maybe jxmapviewer could be re-relased with class files version 50 and not 51?

What's the best way to rebuild current source?

**Comment by msteiger** [comment]
Thanks for reporting - I will have a look asap.

**Comment by matthiasblaesing** [comment]
Maybe this helps (I'm no expert for gradle, but when preparing a patch I also needed to build):

Basic idea:
- signing is only enable when the necessary config is present
- the sonatype values are set to "unset" if they are not present

with these changes a:

<code>gradle clean build install</code>

installs the artifact into the local maven repository.

Modified build.grade:
```
# This patch file was generated by NetBeans IDE
# It uses platform neutral UTF-8 encoding and \n newlines.
--- HEAD
+++ Modified In Working Tree
@@ -1,11 +1,13 @@
 apply plugin: 'java'
 apply plugin: 'maven'
+if(hasProperty("signing.keyId") && hasProperty("signing.password") && hasProperty("signing.secretKeyRingFile")) {
 apply plugin: 'signing'
+}
 apply plugin: 'eclipse'
 apply plugin: 'idea'
 
 group = 'org.jxmapviewer'
-version = '2.0'
+version = '2.1-SNAPSHOT'
 
 repositories {
     mavenCentral()
@@ -15,6 +17,11 @@
 	compile group: 'commons-logging', name: 'commons-logging', version: '1.1.1'
 }
 
+if (! (hasProperty('sonatypeRepoURI') && hasProperty('sonatypeUsername') && hasProperty('sonatypePassword'))) {
+    sonatypeRepoURI = 'unset'
+    sonatypeUsername = 'unset'
+    sonatypePassword = 'unset'
+}
 
 uploadArchives {
     repositories {
@@ -89,6 +96,8 @@
     archives javadocJar
 }
 
+if(hasProperty("signing.keyId") && hasProperty("signing.password") && hasProperty("signing.secretKeyRingFile")) {
 signing {
     sign configurations.archives
 }
+}
```

Maybe this help.

Matthias

**Comment by saper** [comment]
Have a look at https://github.com/msteiger/jxmapviewer2/pull/37 - it works now for me this way (and I get JDK6 class files version 50.0).

I hope you can sign and upload the binary if you provide the properties?

**Comment by msteiger** [comment]
Sorry for the inconvenience about the `2.0`. I will merge your changes and upload version `2.1` to Maven Central.

About the Java6 compatibility: I believe that with the source/target flags, it should run with Java6 even if it is compiled with JDK 8.

**Comment by saper** [comment]
It is recommended to change boot classpath to JRE6 (and also `-extdirs` if other jars from the `ext` are used), because if JDK 7 library features are used by accident, the code compiles silently - it only fails at runtime when started with JRE 6. That's why I put this optional, non-standard variable `JRE6_HOME` in the #37. You can call it whatever you want :)

Netbeans for example solves this problem differently: one should have full target JDK available installed in parallel and that one is invoked, so even `-source` and `-target` flags aren' t needed. I think this is an overkill, so it's simpler just to point to the JRE `rt.jar` instead via the environment (or whatever suits you, really - it can be set in the project properties or somewhere else, just needs a name).

**Comment by msteiger** [comment]
Good point. I used JDK 6_45 to compile the code, signed it and released it as version 2.1 to Maven Central. Everything should be fixed now .Does it solve the issues you had with 2.0?

**Comment by saper** [comment]
Thanks! Now things are perfect - both binary version from maven as well as .jar built with gradle from source compile without warnings when using JDK 1.6 compatibility.
